autorest/azure/azure.go

```diff
@@ -50,8 +51,8 @@ type RequestError struct {
 
 // Error returns a human-friendly error message from service error.
 func (e RequestError) Error() string {
-	return fmt.Sprintf("azure: Service returned an error. Code=%q Message=%q Status=%d",
-		e.ServiceError.Code, e.ServiceError.Message, e.StatusCode)
+	return fmt.Sprintf("autorest/azure: Service returned an error. Code=%q Message=%q Details=%v Status=%v",
+		e.ServiceError.Code, e.ServiceError.Message, e.ServiceError.Details, e.StatusCode)
```

Comment:
this `Error()` function seemed a bit unnecessary. It is almost as same as `ServiceError`

What is the difference of `RequestError` and `ServiceError` again?